pkg/agent/controller/networkpolicy/packetin.go

```diff
@@ -278,17 +271,62 @@ func (c *Controller) rejectRequest(pktIn *ofctrl.PacketIn) error {
 	binary.BigEndian.PutUint32(icmpData[:ICMPUnusedHdrLen], 0)
 	copy(icmpData[ICMPUnusedHdrLen:], ipHdr[:ipHdrLen+8])
 	return c.ofClient.SendICMPPacketOut(
-		srcMAC.String(),
-		dstMAC.String(),
+		srcMAC,
+		dstMAC,
 		srcIP,
 		dstIP,
 		inPort,
-		-1,
+		outPort,
 		isIPv6,
 		icmpType,
 		icmpCode,
 		icmpData,
-		true)
+		packetOutType)
+}
+
+// isServiceTraffic uses ServiceEPStateField to decide if this is a Service
+// traffic, when AntreaProxy is enabled. And when AntreaProxy is disabled, if the
+// destination IP of the reject response packet is on the local Node, but the
+// source MAC is antrea-gw's MAC, then this is a rejection of Service traffic.
+// Because this response is heading to kube-proxy.
```

Comment:
I find the comment quite confusing (the AntreaProxy disabled case).
If this function is meant to be used only for the reject feature, maybe define it as a closure in `rejectRequest` for clarity.

Reply:
Good point. Updated comment and defined as closure.